docs/products/clickhouse/concepts/service-architecture.rst

```diff
@@ -0,0 +1,93 @@
+==========================================
+
+Aiven for ClickHouse® is implemented as a multi-master cluster where the replication of data is managed by ClickHouse itself, the replication of schema and users is managed through ZooKeeper, and data backup and restoration is managed by Astacus. Discover a technical design behind the Aiven for ClickHouse®: its structure and operation. This article helps you understand how Aiven for ClickHouse is set up and configured to work as a managed service.
+
+Deployment modes
+----------------
+
+Aiven for ClickHouse can be deployed either as a single node, a single shard of three nodes, or multiple shards of three nodes each.
+
+* With a single shard, all data is present in all servers and all servers can be used for reads and writes (no main server, leader, or replica).
+* With multiple shards, the data is split between all shards, the data of each shard is present in all nodes of the shard.
+* With multiple nodes, the service is exposed as a high-availability group.
+
+Each Aiven for ClickHouse service is exposed as a single server URL pointing to all servers with connections going randomly to any of the servers. Aiven for ClickHouse is responsible for replicating the writes between the servers.
+For synchronizing critical low-volume information between servers, Aiven for ClickHouse relies on :ref:`ZooKeeper <zookeeper>`, which runs on each ClickHouse server.
+
+Coordinating services
+---------------------
+
+Each Aiven for Clickhouse node runs Clickhouse, ZooKeeper, and Astacus.
+
+.. _zookeeper:
+
+ZooKeeper
+'''''''''
+
+ZooKeeper is responsible for coordination and synchronization across the nodes: one process per node, accessible only from within the cluster.
+
+* By Clickhouse's replicated :ref:`database engine <replicated-database-engine>`, to replicate database changes across the cluster (CREATE, UPDATE or ALTER TABLE)
+* By Clickhouse's ReplicatedMergeTree :ref:`table engine <replicated-table-engine>`, to coordinate replicating table data across the cluster. Data itself is not written to ZK, it is transferred directly between Clickhouse servers
+* By Clickhouse for Replicated Access Entity Storage, this is storage of Users, Roles, Quotas, Row Policies for the whole cluster, in ZooKeeper so that all cluster nodes have the same view of these entities. This storage was developed at Aiven and is now part of the upstream Clickhouse.
+
+Astacus
+'''''''
+
+Astacus is an open-source project originated at Aiven that coordinates backups of cluster databases, including ClickHouse.
+
+Data architecture
+-----------------
+
+* Aiven for ClickHouse enforces full schema replication: all databases, tables, users & grants are the same on all nodes.
+* Aiven for ClickHouse enforces full data replication: table rows are the same on all nodes within a shard.
+* Astacus does most of the backup & restore operations.
+
+.. image:: /images/products/clickhouse/data-architecture.png
```

Comment:
This image is not representative of our current ClickHouse,  I think this should be removed for now.
(And maybe add it later once it's updated).